Gems/Atom/RHI/Vulkan/Code/Source/RHI/RenderPassBuilder.cpp

```diff
@@ -293,15 +295,28 @@ namespace AZ
                 default:
                     AZ_Assert(false, "ScopeAttachmentUsage %d is invalid.", attachment.m_usage);
                 }
-            }            
+            }
+
+            if ((m_subpassCount > 1) && (m_subpassCount == m_renderpassDesc.m_subpassCount))
+            {
+                const auto subpassDependenciesPtr = SubpassDependenciesManager::GetInstance().GetSubpassDependencies(scope.GetId());
+                AZ_Assert(subpassDependenciesPtr != nullptr, "Subpass Dependencies for scope [%s] do not exist.", scope.GetId().GetCStr());
+                AZ_Assert(subpassDependenciesPtr->m_subpassCount == m_subpassCount,
+                    "Subpass Dependencies for scope [%s] was created for %u subpasses, but this Render Pass is being created with %u subpasses",
+                    scope.GetId().GetCStr(), subpassDependenciesPtr->m_subpassCount, m_subpassCount);
+                subpassDependenciesPtr->CopySubpassDependencies(m_renderpassDesc.m_subpassDependencies);
+            }
 
+            // TODO: The following block should NOT exist because Subpasses in Vulkan
```

Comment:
I think tile shaders in metal let you use buffers to allow grouping a compute and raster pass. Although we dont have this support for the metal backend atm.

Reply:
Ok, will remove that comment.